File: Backend/OguzlarBelediyesi.Domain/Entities/Operations/MunicipalUnit.cs
using System;
using System.Collections.Generic;
using OguzlarBelediyesi.Domain.Entities.Common;

namespace OguzlarBelediyesi.Domain;

public sealed record UnitStaff(
    string Name,
    string Title,
    string? ImageUrl = null);

public sealed class MunicipalUnit : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string? Content { get; set; }
    public string Icon { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public List<UnitStaff>? Staff { get; set; }
}

File: BackendUI/src/app/models/municipal-unit.ts
export interface UnitStaff {
    name: string;
    title: string;
    imageUrl?: string | null;
}

export interface MunicipalUnit {
    id: string;
    title: string;
    content?: string;
    icon: string;
    slug: string;
    staff?: UnitStaff[] | null;
}
